### PR TITLE
Refactor logging and fix issues

### PR DESCRIPTION
**Pull request by liangxin1300**
Commit 45742f1 is for #12 , refactor logging using dictConfig;
Commit aec065d is for #6 , to make sure normal users be able to use -h/--help
Commit d908fdf57a565ae35c28be13e8534a74eb186fee is for #7, to make sure this script can only be executed as root

### CODE REVIEW

**Comment by liangxin1300** [comment]
Hi @tomschr , please review it if you have time:)

Thanks!

File: cluster_preflight_check/main.py [hunk 1]
```diff
@@ -8,6 +8,9 @@
 import time
 import threading
 import logging
+logger = logging.getLogger('cpc')
+logging.basicConfig(level=logging.DEBUG)
```
[the remark this answers is not in the document]
Comment (liangxin1300) [reply cluster_preflight_check/main.py]: Changed
Thanks!

File: cluster_preflight_check/main.py [hunk 2]
```diff
@@ -336,44 +340,65 @@ def setup_logging(context):
...
     major work flow
     '''
-    setup_logging(context)
     setup_basic_context(context)
     parse_argument(context)
+    if not utils.is_root():
+        logging.fatal("{} can only be executed as user root!".format(context.name))
+        sys.exit(1)
+    if not os.path.exists(context.var_dir):
+        os.mkdir(context.var_dir)
```
[the remark this answers is not in the document]
Comment (liangxin1300) [reply cluster_preflight_check/main.py]: Changed
Thanks!

File: cluster_preflight_check/main.py [hunk 3]
```diff
@@ -8,6 +8,9 @@
 import time
 import threading
 import logging
+logger = logging.getLogger('cpc')
```
[the remark this answers is not in the document]
Comment (liangxin1300) [reply cluster_preflight_check/main.py]: Thanks for suggestions!

File: cluster_preflight_check/utils.py [hunk 4]
```diff
@@ -6,6 +6,7 @@
 import socket
 import json
 import logging
+logger = logging.getLogger('cpc')
```
[the remark this answers is not in the document]
Comment (liangxin1300) [reply cluster_preflight_check/utils.py]: Thanks!

File: cluster_preflight_check/main.py [hunk 5]
```diff
@@ -336,44 +340,65 @@ def setup_logging(context):
...
+            },
+            'stream': {
+                'class': 'logging.StreamHandler',
+                'formatter': 'stream_formatter'
+            }
+        },
+        'loggers': {
+            'cpc': {
+                'handlers': ['null', 'file', 'stream'],
+                'propagate': False
```
[the remark this answers is not in the document]
Comment (liangxin1300) [reply cluster_preflight_check/main.py]: Thanks!

File: cluster_preflight_check/utils.py [hunk 6]
```diff
@@ -56,13 +57,12 @@ def now(form="%Y/%m/%d %H:%M:%S"):
 
 
 def msg_raw(level, msg, to_stdout=True):
-    from . import main
```
[the remark this answers is not in the document]
Comment (liangxin1300) [reply cluster_preflight_check/utils.py]: Thanks for this!
And I changed it, maybe more generic:
```
@contextmanager
def manage_hander(_type, keep=True):
    try:
        handler = get_handler(logger, _type)
        if not keep:
            logger.removeHandler(handler)
        yield
    finally:
        if not keep:
            logger.addHandler(handler)
```

**Comment by liangxin1300** [comment]
Hi @tomschr , for all of your great suggestions, thank you!:)

**Comment by tomschr** [comment]
@liangxin1300: Happy to help! :+1:

**Pull request merged**